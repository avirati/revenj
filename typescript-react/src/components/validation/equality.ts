import { Eq, ValidatorCreator } from './interfaces';
import * as operators from './operators';
import { validatorCreatorFactory } from './validatorCreatorFactory';

////////////////////////////////////////////////////////////////////////////////
// Equality validators                                                        //
////////////////////////////////////////////////////////////////////////////////
interface IEqValidatorCreatorFactory {
  <T extends Eq>(value: T): ValidatorCreator<T, T, void, void>;
}

export const isEqualCreator: IEqValidatorCreatorFactory = <T>(value: T) =>
  validatorCreatorFactory<T, T, void, void>({
    getValidatorBaseValue: () => value,
    getValidatorErrorMessage: (base) => `Value must be equal to ${base}.`,
    operator: operators.eq,
  });
