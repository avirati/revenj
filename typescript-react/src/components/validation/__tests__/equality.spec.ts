import { BigNumber } from 'bignumber.js';

import { isEqualCreator } from '../equality';

describe('validation', () => {
  describe('isEqualCreator', () => {
    type AThing = [number, number, string];
    interface ISomethingElse { x: number; y: { z: string }; }

    const isEqual = <T>(val: T) => isEqualCreator(val)();

    it('should return undefined if input value is a multiple of base value', () => {
      expect(isEqual(44)(44)).toBeUndefined();
      expect(isEqual('asdf')('asdf')).toBeUndefined();
      expect(isEqual(false)(false)).toBeUndefined();
      expect(isEqual(undefined)(undefined)).toBeUndefined();
      expect(isEqual(null)(null)).toBeUndefined();
      expect(isEqual(new BigNumber(-7))(new BigNumber(-7))).toBeUndefined();
      expect(isEqual<AThing>([1, 2, '3'])([1, 2, '3'])).toBeUndefined();
      expect(isEqual<ISomethingElse>({ x: 2, y: { z: 'wat' } })({ x: 2, y: { z: 'wat' } })).toBeUndefined();
    });

    it('should return an error message if input value is not a multiple of base value', () => {
      expect(isEqual(44)(13)).toBeDefined();
      expect(isEqual(44)(13)).toMatchSnapshot();
      expect(isEqual('asdf')('aasdf')).toBeDefined();
      expect(isEqual('asdf')('aasdf')).toMatchSnapshot();
      expect(isEqual(false)(true)).toBeDefined();
      expect(isEqual(false)(true)).toMatchSnapshot();
      expect(isEqual(new BigNumber(-7))(new BigNumber(7))).toBeDefined();
      expect(isEqual(new BigNumber(-7))(new BigNumber(7))).toMatchSnapshot();
      expect(isEqual<AThing>([1, 2, '3'])([3, 4, 'asd'])).toBeDefined();
      expect(isEqual<AThing>([1, 2, '3'])([3, 4, 'asd'])).toMatchSnapshot();
      expect(isEqual<ISomethingElse>({ x: 0, y: { z: 'hmmm' } })({ x: 2, y: { z: 'wat' } })).toBeDefined();
      expect(isEqual<ISomethingElse>({ x: 0, y: { z: 'hmmm' } })({ x: 2, y: { z: 'wat' } })).toMatchSnapshot();
    });
  });
});
