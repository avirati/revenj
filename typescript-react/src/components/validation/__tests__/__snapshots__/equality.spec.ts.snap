// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 1`] = `"Value must be equal to 44."`;

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 2`] = `"Value must be equal to asdf."`;

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 3`] = `"Value must be equal to false."`;

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 4`] = `"Value must be equal to -7."`;

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 5`] = `"Value must be equal to 1,2,3."`;

exports[`validation isEqualCreator should return an error message if input value is not a multiple of base value 6`] = `"Value must be equal to [object Object]."`;
